discovery/file/file_test.go

```diff
@@ -90,22 +91,21 @@ func testFileSD(t *testing.T, prefix, ext string, expect bool) {
 		t.Fatal(err)
 	}
 
-	// File is written with the config so stop draining the discovery channel.
-	// It needs to be before the file closing so that fsnotify triggers a new loop of the discovery service.
+	// Config file is ready so stop draining the discovery channel.
+	// It contains two target groups.
 	close(fileReady)
 	<-drainReady
-	newf.Close()
+	newf.WriteString(" ") // One last meaningless write to trigger fsnotify and a new loop of the discovery service.
```

Comment:
`newf.Sync()` would do the trick no? And why doesn't `Close()` work here, it also syncs the data to disk, AFAIK.

Reply:
I tried `newf.Sync()` and it doesn't trigger a new fsnotify signal